boards/blackpill-128kib/Kconfig

```diff
@@ -17,4 +17,11 @@ config BOARD_BLACKPILL_128KIB
 
     select HAS_HIGHLEVEL_STDIO
 
+# HACK: This is added due to the make resolution
+# make will select timer backend, probably due to the USBUS
+# and kconfig cannot select if something is already selected like make
+choice CHOICE_ZTIMER_MSEC_BACKEND
+    default ZTIMER_MSEC_BACKEND_TIMER if MODULE_STDIO_UART_RX
```

Comment:
If this is due to USBUS why does it depend on uart rx?

Reply:
USBUS_STDIO eventually selects this due to STDIN.  I didn't dig down as this is a hack.  I just noticed that without this conditional `examples/hello-world` fails.  That somewhat clued me in that is is to do with the STDIN stuff.

I am hoping that this will eventually get cleaned up nicely making the hack obsolete.  Maybe something @jue89 would be interested in taking on :smiley:

Reply:
Nevermind it makes it fail again...

Reply:
Oh, I'm far away from obtaining the black belt in Kconfig modeling :D

Reply:
> Oh, I'm far away from obtaining the black belt in Kconfig modeling :D

In case you're interested ;) besides the ultimate word in this https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html I found useful https://docs.zephyrproject.org/latest/build/kconfig/tips.html 

Many of the recommendations were taken early on in the design, many others had to be gradually discarded (e.g., `select`). Still it's a good read to understand the trade-offs that one makes when using one approach instead of the other.